addons/l10n_se/models/luhn.py

```diff
@@ -0,0 +1,32 @@
+from odoo import models
+
+class luhn(models.AbstractModel):
+    """ Method calls works as Luhn 0.2.0 library. """
```

Comment:
I will try to explain the difference, just need to take a deeper look at the code. It might even be that it's broken in stdnum as I stated earlier.

Here is the code from the library stdnum latest version:

```
def checksum(number, alphabet='0123456789'):
    """Calculate the Luhn checksum over the provided number. The checksum
    is returned as an int. Valid numbers should have a checksum of 0."""
    n = len(alphabet)
    number = tuple(alphabet.index(i)
                   for i in reversed(str(number)))
    return (sum(number[::2]) +
            sum(sum(divmod(i * 2, n))
                for i in number[1::2])) % n
```
So lets do a try run:

If you do ask checksum:

My function: 6 will give 7 as checksum.
stdnum: 6 gives 6.... 

Ohh hey wait a minute.. I might have done this wrong.. So lets triple check it with OCR Checkup.... 

Can be found here: 
https://www.bankgirot.se/en/services/incoming-payments/bankgiro-receivables/ocr-reference-control/

Does 67 work or 66.. In this case 67 is correct and not 66.. So basicly in this case the luhn algorithm isn't working... Lets try some other cases:

stdnum: 5 gives 5
My luhn: 5 gives 9.. 
OCR Ref: 59 is correct.

Ok. I bet I didn't get the point out the first time. The stdnum luhn algorithm is broken/bugged and is not working in this case. How do I know this. Well I did a analysis of it earlier and found the reson for it together and got it verified by one of Odoo:s programmers...

I will try to show this in some more cases:
stdnum: 73 gives 8
my+ocr ref says: 73 gives 7

stdnum: 114 gives 7
my+ ocr: 114 gives 9

stdnum: 124 gives 9
my + ocr: 124 gives 8

stdnum: 134 gives 1
my + ocr: 134 gives 7

sdnum 144 gives 3
my + ocr: 144 gives 6

So the stdnum implementation isn't working as it is suppose or they have implemented a specific version as a general version. However this doesn't matter it's not working in this cases. It might also be the case that the luhn algorithm used in sweden is implemended in another way then normaly. However as I stated before my version is working for Sweden. 

You might notice that I have rewritten the function between this time and the earlier version released for l10n_se project. The reason for this is that I found an inconsistancy in one of the functions that was used in re. This made the function calculate it wrong in some specific cases. The new function however solves this and doesn't depend on re.

Reply:
@daniel-stenlov Thanks for the explanation.  The point is to have this in a comment in the code. (can be short, referring the link of bankgirot and telling that stdnum does not give the same results)

Reply:
I forgot to say that it's the same thing with the other cases of luhn used in sweden like accountnumber and so on. It might be working for Personnummer or organisational numbers since they have added their own functions for those even if the calculation is the same. I haven't tested those in this case.